swf/src/read.rs

```diff
@@ -1576,16 +1576,34 @@ impl<'a> Reader<'a> {
                 FillStyle::Color(color)
             }
 
-            0x10 => FillStyle::LinearGradient(self.read_gradient(shape_version)?),
+            0x10 => {
```

Comment:
In order to avoid code duplication, it might be worth to unify all 3 gradient cases into a single one using `0x10..=0x13`.